nested configurations in extension: sub-property is seen as nested entity.

### Describe the bug

The nightly builds for `quarkus-artemis` against the `main` of `quarkus` fail since 2023-01-04, see e.g. [here (`github.com`)](https://github.com/quarkiverse/quarkus-artemis/actions/runs/7664166199).

We have a similar configuration scheme as data sources have: we allow named connection factories, with one default connection factory. Looking at the stack trace of the build we find this stack trace:
```
java.lang.RuntimeException: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.test.junit.QuarkusTestExtension.throwBootFailureException(QuarkusTestExtension.java:643)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:727)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:285)
	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:252)
	at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:610)
	at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:660)
	... 1 more
Caused by: java.lang.IllegalStateException: Configuration devservices: url is not set and devservices is activated. This is a bug. Please report it.
	at io.quarkus.artemis.core.runtime.ArtemisUtil.validateIntegrity(ArtemisUtil.java:30)
	at io.quarkus.artemis.jms.runtime.ArtemisJmsRecorder.getConnectionFactoryProducer(ArtemisJmsRecorder.java:32)
	at io.quarkus.deployment.steps.ArtemisJmsProcessor$configure1243796012.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.ArtemisJmsProcessor$configure1243796012.deploy(Unknown Source)
	... 8 more
```
This means that `devservices` is recognized as named connection factory, not as property of the default configuration.

Further details can be found on [the PR that should upgrade `quarkus` to `3.7.0`  in `quarkus-artemis` (`github.com`)](https://github.com/quarkiverse/quarkus-artemis/pull/382).

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -51,7 +51,7 @@
         <microprofile-lra.version>2.0</microprofile-lra.version>
         <microprofile-openapi.version>3.1.1</microprofile-openapi.version>
         <smallrye-common.version>2.2.0</smallrye-common.version>
-        <smallrye-config.version>3.5.2</smallrye-config.version>
+        <smallrye-config.version>3.5.4</smallrye-config.version>
         <smallrye-health.version>4.1.0</smallrye-health.version>
         <smallrye-metrics.version>4.0.0</smallrye-metrics.version>
         <smallrye-open-api.version>3.9.0</smallrye-open-api.version>

